demo/admin/src/products/future/generated/ProductForm.gql.tsx

```diff
@@ -13,6 +13,14 @@ export const productFormFragment = gql`
             id
             title
         }
+        manufacturerCountry: manufacturer {
+            id: addressAsEmbeddable {
+                country
+            }
+            label: addressAsEmbeddable {
+                country
+            }
+        }
```

Comment:
I don't think this is how I'd write this code if I'd write it by hand. Isn't this feature a workaround cause by the unlikely usage of async selects in the manufacturers example?

Reply:
what do you mean? query different attributes of the same object for different inputs or how the data is nested?

I think initial value/label for filter-fields will most of the time come from the same object as the filtered-field/entity. Because it's naturally some attribute of the filtered-entity required to filter them. (e.g. brand if selecting a car-model) Saving the filter-value as a separate field or providing the filter-value of the filtered-entity on the same level as the filtered-entity (e.g. by implementing a field-resolver fetching the car-model, returning only brand) should be the exception...

the weird stuff of manufacturerCountry is double-nesting (maybe even not, e.g. model.brand.id/label vs manufacturer.address.country/country) and probably much more reusing country-field as label and id.

Reply:
If I did this by hand, I'd write the following query:

```
export const productFormFragment = gql`
    fragment ProductFormDetails on Product {
        manufacturer {
            id
            name
            addressAsEmbeddable {
                country
            }
        }
    }
`;
```

and the following initial values code:

```ts
const initialValues: Partial<FormValues> = useMemo<Partial<FormValues>>(() => {
    const filteredData = data ? filterByFragment<ProductFormManualFragment>(productFormFragment, data.product) : undefined;
    /*...*/
    return {
        ...filteredData,
        image: rootBlocks.image.input2State(filteredData.image),
        manufacturerCountry: filteredData.manufacturer
            ? {
                    id: filteredData.manufacturer?.addressAsEmbeddable.country,
                }
            : undefined,
    };
}, [data]);
```

So without the additional `manufacturerCountry` at the "top level" of the query

Reply:
> So without the additional manufacturerCountry at the "top level" of the query

I just had a look at the handmade form: Why did decide to implement this with the second "field"?